Use regex for web routing

nexus4880 opened the pull request:
I was implementing some more routes when I found our dynamic route matching system was flawed.

I have switched all existing routes to use regular expressions. All tests pass locally and I have started the game and gotten into a raid without any issues.

nexus4880 committed:
Initial commit

nexus4880 committed:
Updated expression to match es-mx

seionmoya reviewed (changes requested):
I am currently not willing to merge the PR in it's current state. A couple of things that need to change:
- base constructor: one for string, one for regex
- the derived class using static paths uses the string base constructor
- the derived class using dynamic paths uses the regex base constructor and stores the regex in compileregenerated attribute (See `AccountValidation` class)
- the quality of the regex needs to improve (make it as dead-simple as possible)

seionmoya commented:
@Lacyway can you look into simplifying / improving the quality of the regex used for the locale controllers in this PR?

nexus4880 committed:
Added WebController(Regex) and WebController(string)
Updated existing endpoints to use string constructor if static
Updated existing endpoints to use compiler generated regex if dynamic

nexus4880 committed:
Updated format on note

seionmoya committed:
Fix AccountValidation regex

seionmoya committed:
Update LocaleController.cs

seionmoya committed:
Update MenuLocaleController.cs

seionmoya committed:
Update HttpController.cs

seionmoya committed:
Update WsController.cs

seionmoya committed:
Merge branch 'dev' into use-regex-for-web-routing

seionmoya committed:
Update HttpController.cs

nexus4880 committed:
Merge branch 'project-fika:dev' into use-regex-for-web-routing
